### PR TITLE
Support upgrade test between any two releases.

### PR DESCRIPTION
**Pull request by yusuoh**
1. Add target version and other options.
2. Support sidecar upgrade

### CODE REVIEW

File: tests/e2e/framework/framework.go [hunk 1]
```diff
@@ -84,13 +84,15 @@ func InitLogging() error {
 	return nil
 }
 
-// NewTestConfig creates a full config will all supported configs.
-func NewTestConfig(testID, baseVersion string) (*CommonConfig, error) {
+// NewCommonConfigWithBaseVersion creates a new CommonConfig with the specified
+// version of Istio. If baseVersion is empty, it will use the local head
```
Comment (chxchx) [inline tests/e2e/framework/framework.go]: nit: baseVersion -> version to be consistent with function param

**Comment by chxchx** [comment]
/lgtm

Now that you are at this, do you mind taking a look at https://github.com/istio-releases/daily-release/blob/master/prow/daily-e2e-rbac-no_auth-skew.sh (which is at istio-releases/daily-release, a different repo and different org)? It is a version skew test that we run for daily release, which test if current version of control plane works with previously released proxy sidecars. It has been failing for a while ([example run](https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio-releases_daily-release/416/daily-e2e-rbac-no_auth-skew/19/)) and I was not sure if the failure was due to incompatibility or I was using the flags wrong.  

It is really similar to what the upgrade test is doing. Perhaps I should just replace it with a call to this make target?

File: tests/e2e/framework/framework.go (hunk 1 above)
Comment (chxchx) [reply tests/e2e/framework/framework.go]: I was referring to the function comments tho. Updating the function name is also superb.

**Comment by yusuoh** [comment]
Yes. I can take care of that version skew test. Is that repo solely used for release? How is it related to the istio.io/test-infra repo?

**Comment by chxchx** [comment]
Yes that repo is used to run qualification tests on daily releases.

Regarding my earlier comment, @yusuoh and I talked offline and confirmed that the failure on the skew test is expected due to incompatible API change.

**Comment by yusuoh** [comment]
Hi @sebastienvas , could you please approve this PR?

**Comment by ldemailly** [comment]
if you see weird vendor errors in the future please alert me right away

thanks

**Pull request merged**